docs/src/content/docs/guides/sidebar.mdx

```diff
@@ -520,6 +520,65 @@ Browsing the documentation in Brazilian Portuguese will generate the following s
 In multilingual sites, the value of `slug` does not include the language portion of the URL.
 For example, if you have pages at `en/intro` and `pt-br/intro`, the slug is `intro` when configuring the sidebar.
 
+### Internationalization with badges
+
+For [badges](#badges), the `badge` property or the `text` when using the object form can be a string, or for multilingual sites, an object with values for each different locale.
+When using the object form, the keys must be [BCP-47](https://www.w3.org/International/questions/qa-choosing-language-tags) tags (e.g. `en`, `ar`, or `zh-CN`):
+
+```js {11-14,19-22}
+starlight({
+	sidebar: [
+		{
+			label: 'Constelações',
+			translations: {
+				'pt-BR': 'Constelações',
+			},
+			items: [
+				{
+					slug: 'constellations/andromeda',
+					badge: {
+						en: 'New',
+						'pt-BR': 'Novo',
+					},
+				},
+				{
+					slug: 'constellations/scorpius',
+					badge: {
+						text: {
+							en: 'Outdated',
+							'pt-BR': 'Descontinuado',
+						},
+						variant: 'caution',
+					},
+				},
+			],
+		},
+	],
```

Comment:
Awesome stuff! Happy to add this.

I wonder if it would be simpler to only support localized strings using the `text` property though? I find it a bit weird to have two config options that are objects but that mean very different things depending on what the keys are, i.e. one object is “record of locales to labels” the other is “precisely typed config object with known keys”. (Without testing, I’d guess this also interferes with IDE autocomplete?)

Also, imagine this badge:
```js
badge: {
  en: 'New',
  fr: 'Nouveau',
}
```
If you decide you want to make it a `success` variant, this doesn’t work:
```diff
badge: {
  en: 'New',
  fr: 'Nouveau',
+  variant: 'success',
}
```
So overall, I think I’m in favour of a single syntax rather than two. What do you think?

Reply:
Ny initial implementation was what you described, then I somehow managed to convince myself that if you were using the shorthand notation, you would prefer to use that too with translated labels. I ended up not liking it personally but couldn't decide if I should remove it or not.

If you're having the same feeling too, I'll be more than happy to remove it (and it'll make the implementation simpler). I'll update the PR accordingly. Thanks again for the great input 🙌

Reply:
Haha, yeah, going from `badge: 'New'` to `badge: { text: { en: 'New' } }` feels superficially like a lot, but I think in practice it’s OK and even it is slightly verbose, it definitely is clearer to explain/understand. Plus once you have a couple of locales the difference is kind of not so bad, e.g.
```js
badge: { text: { en: 'New', fr: 'Nouveau', de: 'Neu', it: 'Nuovo' } }
// vs
badge: { en: 'New', fr: 'Nouveau', de: 'Neu', it: 'Nuovo' }
```

Reply:
I second having only one config option with locales inside `text`. It's a key longer but a very small price to pay to have **localized** badges 💜.

Reply:
Thanks you both, super appreciated. I have updated the PR accordingly, definitely an improvement 👍